src/process_utility.c

```diff
@@ -834,13 +891,18 @@ process_rename_table(Cache *hcache, Oid relid, RenameStmt *stmt)
 
 		if (NULL != chunk)
 			ts_chunk_set_name(chunk, stmt->newname);
+
+		return false;
 	}
-	else
-		ts_hypertable_set_name(ht, stmt->newname);
+
+	ts_hypertable_set_name(ht, stmt->newname);
+
+	process_add_hypertable(args, ht);
+	return true;
```

Comment:
Thats a bit different here. Return value here says that is should be handled in `process_rename`. Since we are working under cache pin taken in `process_rename` we should call processUtility after cache release and then pass it down. It should crash on assert in `hypertable_cache_create_entry`, because relation and and one that in our schema will not match.